### ISSUE TITLE
How are cls_score and bbox_pred resized to match labels and bbox_targets

### ISSUE THREAD

**Issue by AlexTS1980** [jasjeetIM/Mask-RCNN#13]
In the training phase the number of predicted RoIs + scores is determined by hyperparameters like  `POST_NMS_ROIS_TRAINING`. Caffe inits layers to (1,21) or (1,84) and then they are resized to match the number of predictions, e.g. `(1000, 21)` and `(1000,84)`. This is done in custom layers, and I don't have a problem with that. At the same time, in order to get the loss, layers like `cls_score, bbox_pred, fc6/7`, are also resized to match the size (while keeping the number of weights the same). I don't see how Caffe does it out of the box, but I can't find the wrapper for that. Could anyone help?
